### PR TITLE
added option to add the folder to the ignored ones in the popup

### PR DESCRIPTION
**Pull request by jonnytest1**
with this there will be another option next to "Run Setup Wizard" called "Ignore Folder" which when pressed adds the corresponding workspace fodler directly to the ignored folders in the workspace config

i was thinking about , searching the complete workspace for jest files and not enable it at all or maybe a whitelist but i think its better for the user to get notified when a workspace will not be covered by jest and to consciously decide

### CODE REVIEW

**Comment by jonnytest1** [comment]
could it be you have conflicting "prettier" and "eslint - prettier" configuration ? oO

**Comment by jonnytest1** [comment]
one more hint : since prettier has a rule for line endings to be LF ones i think changing 
`* text=auto ` to `* text=false` in the .gitattributes files makes more sense 🤔

File: src/JestExt/core.ts [hunk 1]
```diff
@@ -144,6 +145,14 @@ export class JestExt {
     };
   }
 
+  private setupIgnoreAction(folder: string): messaging.MessageAction {
```
Comment (connectdotz) [inline src/JestExt/core.ts:148]: each JestExt/core corresponding to a workspace, you can find it in `this.extContext.workspace.name`. So we don't need to pass this extra argument around from the listeners to here. It should eliminate some API changes for the exit event.

File: src/JestExt/core.ts [hunk 2]
```diff
@@ -176,7 +185,8 @@ export class JestExt {
             messaging.systemErrorMessage(
               event.error,
               messaging.showTroubleshootingAction,
-              this.setupWizardAction('cmdLine')
+              this.setupWizardAction('cmdLine'),
+              this.setupIgnoreAction(event.workspaceFolder)
```
Comment (connectdotz) [inline src/JestExt/core.ts:189]: I think we should only add the additional action for multi-root workspaces, something like
```
if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1) {
...
}
```

**Review by connectdotz — CHANGES_REQUESTED** [review]
@jonnytest1 thanks for the PR! 

I have a few comments, let me know what you think.

File: src/JestExt/core.ts (hunk 1 above)
Comment (jonnytest1) [reply src/JestExt/core.ts:148]: oh i was looking into that real quick cause this
```
this.session.context.onRunEvent.fire({
        type: 'exit',
        process,
        workspaceFolder: this.session.context.workspace.name,
      }); 
```
was quite the giveaway they're somwhere close (i didnt look further into it when i didnt find a property called session ^^)

gonna refactor to use this
`this.extContext.workspace.name `

File: src/JestExt/core.ts (hunk 2 above)
Comment (jonnytest1) [reply src/JestExt/core.ts]: 🤔 on second thought i do also have single folder workspaces that dont work with jest where disabling it with one click might also be nice

Comment (connectdotz) [reply src/JestExt/core.ts]: for single folder workspace, I think the current activation check should be sufficient; and they can simply disable the extension via vscode if it's truly not applicable. However, multi-root workspace is different:
1. we did not perform the same activation check for each multiroot workspace folder
2. users can't simply disable the extension for a certain workspace folder via vscode, thus the introduction of the option `jest.disabledWorkspaceFolders` for more granular control.

**Comment by connectdotz** [comment]
Looks like the test coverage has been reduced. @jonnytest1, do you mind adding some tests for the code change? Much appreciated.

**Comment by jonnytest1** [comment]
oh lol i can scroll the checks :D

**Review by connectdotz — APPROVED** [review]
Looks good. The test coverage dip is all right as new changes are now indeed tested. 

There are other issues such as double reporting errors but it is not related to this PR, we will address them in another PR.

**Pull request merged**